### PR TITLE
chore: add default upgrade handler for v8.1

### PR DESCRIPTION
**Pull request by damiannolan**
## Description

Adds the default upgrade handler for v8.1.

ref: #5635

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against the correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#pull-request-targeting)).
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/main/docs/build/building-modules/11-structure.md) and [Go style guide](../docs/dev/go-style-guide.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/main/testing/README.md#ibc-testing-package).
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`).
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Provide a [commit message](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) to be used for the changelog entry in the PR description for review.
- [ ] Re-reviewed `Files changed` in the Github PR explorer.
- [ ] Review `Codecov Report` in the comment section below once CI passes.

### CODE REVIEW

File: testing/simapp/upgrades/upgrades.go [hunk 1]
```diff
@@ -107,10 +109,3 @@ func CreateV7LocalhostUpgradeHandler(
 		return mm.RunMigrations(ctx, configurator, vm)
 	}
 }
-
-// CreateV8UpgradeHandler creates an upgrade handler for the ibc-go/v8 SimApp upgrade.
-func CreateV8UpgradeHandler(mm *module.Manager, configurator module.Configurator) upgradetypes.UpgradeHandler {
```
Comment (damiannolan) [inline testing/simapp/upgrades/upgrades.go:112]: This was never needed. Duplicate of `CreateDefaultUpgradeHandler`

**Review by srdtrk — APPROVED** [review]
lgtm

**Review by DimitrisJim — APPROVED** [review]
noice!

**Review by chatton — APPROVED** [review]
LGTM

**Pull request merged**